python/cudf/cudf/core/column/column.py

```diff
@@ -172,7 +172,7 @@ def equals(self, other: ColumnBase, check_dtypes: bool = False) -> bool:
         if check_dtypes:
             if self.dtype != other.dtype:
                 return False
-        return (self == other).min()
+        return (self.fillna(True) == other.fillna(True)).fillna(False).min()
```

Comment:
Why do we need these `fillna`? If there's nulls it would give nulls from the binaryop, which would get filtered out by the `min` / `max` call.

Reply:
Consider the two following series:

```
>>> x = cudf.Series([1,2,None])
>>> y = cudf.Series([1,2,   3])
```
These two are undoubtedly _not_ equal. With this change, `x == y` gives 
```
>>> x == y
0    True
1    True
2    <NA>
dtype: bool
```
And then `(x == y).min()` returns `True`, which is not what we want.